src/OpenApi/src/Extensions/JsonTypeInfoExtensions.cs

```diff
@@ -104,7 +104,9 @@ internal static string GetSchemaReferenceId(this Type type, JsonSerializerOption
         // Generic types become a concatenation of the generic type name and the type arguments
         if (type.IsGenericType)
         {
-            var genericTypeName = type.Name[..type.Name.LastIndexOf('`')];
+            // We need to handle the case where the generic type is a nested type, so we check if the name contains a backtick already.
+            // Example: https://github.com/dotnet/aspnetcore/issues/59092
+            var genericTypeName = type.Name.Contains('`') ? type.Name[..type.Name.LastIndexOf('`')] : type.Name;
```

Comment:
What about calling ```type.Name.LastIndexOf('`')``` first, then checking whether it's value `> -1`? Then it's not two lookups into the string.

Reply:
Good call! I guess I was confused and thought `LastIndexOf` threw, not realizing it actually returns -1. I'll make the change!

Reply:
@martincostello Done! Let me know if you agree with this changed approach and feel free to resolve if so!